### ISSUE TITLE
Default value in algorithm registration needs quotes to be a string

### ISSUE THREAD

**Issue by grallewellyn** [MAAP-Project/ZenHub#919]
"When Jamison registered the algorithm, he had to literally include double-quotes around default values. Unfortunately, although this allowed successful registration, this doesn't play well with the algorithm's script because those quotes end up being part of the actual value passed through when the user does not supply a value.
For example, for our beams input, Jamison had to specify "all" as the default value (literally with quotes) rather than all (no quotes). When the algorithm script receives that default value, it literally contains double quotes, so the algorithm bombs because "all" (with quotes) is not a valid value for the beams field. It should be all (no quotes)."
-Chuck Daniels

**Comment by grallewellyn** [comment]
Instead of adding type checking:
"I'd suggest then, that until that happens, or even when it does happen, but the data type is not provided (I assume that, "if provided in spec" means that it would be optional to specify the type), that the field should be treated as a string.
I don't know how fields are currently handled such that the double quotes are required for default values, but sadly, that simply renders default values unusable, as mentioned above.
It seems strange that since there is currently no check of the data type at the moment that we get this odd behavior. Without any type information, values should all simply be handled as string since that's how all values get passed through to the algorithm's script. In other words, since all algorithms get invoked as commands on the command-line, all arguments are strings, no matter what. Even when type checking occurs first, every input value ends up as a string on the command line, and it is up to each algorithm to parse command-line arguments as necessary.
Bottom line: in the absence of any type checking, don't do anything with input values or their default values. Simply pass them through. This means there should be no need to put quotes around default values, especially since that simply breaks the use of default values altogether. The only workaround for this is for an algorithm's script to strip off double quotes surrounding each argument, which is inconvenient."
-Chuck Daniels

**Comment by chuckwondo** [comment]
This appears to no longer be an issue. When an algorithm's run script is invoked, DPS passes a value for _every_ input defined for the algorithm. When the user does not supply a value for an input, DPS still passes an empty string in the correct argument position.

For example, if an algo takes 3 inputs, but only the first input is supplied by the user, DPS will pass empty strings for the other two:

```plain
run.sh 'user-supplied-value' '' ''
```

Notice that although the 2nd and 3rd values are empty strings, they are recognized as argument values, meaning that the `run.sh` script will still receive 3 arguments, not just 1.
